### ISSUE TITLE
Support GHC nightlies obtained via ghcup

### ISSUE THREAD

**Issue by Bodigrim** [haskell-CI/haskell-ci#665]
It would be great to be able to add a job testing GHC HEAD, presumably using a new `ghcup` feature to download nightlies: https://www.haskell.org/ghcup/guide/#nightlies

**Comment by phadej** [comment]
Apparently `allow-failure` doesn't work on GHA. Why did I think it does. Well, I'm not going to add GHC nightly support then yet, it should be possible to mark it as fallible. 

https://github.com/actions/runner/issues/2347

**Comment by phadej** [comment]
Wait, `continue-on-error: ${{ matrix.experimental }}` may work. Let me try that.

**Comment by phadej** [comment]
The action is marked as passing: https://github.com/haskell-CI/haskell-ci/actions/runs/5754864293, but the PR&commit are not https://github.com/haskell-CI/haskell-ci/pull/671

That is silly, and there is issue https://github.com/orgs/community/discussions/15452 about it.

**Comment by phadej** [comment]
This is also blocked on https://github.com/haskell/ghcup-metadata/issues/149#issuecomment-1807774019 

> I totally agree artifacts from the nightly pipeline are not a thing anybody should rely on.

though in `haskell-ci` usage we would ever grab only the latest, which is hopefully nor purged. Apparently it happened at least once, that nightly pipeline was unstable enough, so the latest successful one was purged.

In other words, until GHC folds themself explicitly says (e.g. somewhere in GHC wiki) that the nighties can be used for (automated) testing (which I think they did, so I'm getting mixed messages), only then I'll reconsider using nightlies.

Until then I close this issue, as this is truly blocked.

**Comment by hasufell** [comment]
Yes, nightlies have been a failure. GHC developers have not been able to provide a reliable infrastructure for them.

**Comment by chreekat** [comment]
> so I'm getting mixed messages...

@phadej I should specify that my opinions I've shared here on this ticket only represent my personal views. :P I was unhappy with GHC Nightlies as soon as I heard about how they were implemented.

I've tried to improve the situation by better clarifying their (lack of) reliability. See https://gitlab.haskell.org/ghc/ghc/-/wikis/GHC-Status#2-ghc-nightlies.

Dashboard with status: https://grafana.gitlab.haskell.org/d/ab109e66-a8a1-4ae9-b976-40e2dfe281ab/availabilitie-of-ghc-nightlies-via-ghcup?orgId=2&from=1681980144490&to=1697791344490&refresh=1d
